fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -886,6 +927,14 @@ ACTOR Future<Void> preresolutionProcessing(CommitBatchContext* self) {
 	if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST) {
 		self->writtenTagsPreResolution = self->getWrittenTagsPreResolution();
 	}
+
+	if (SERVER_KNOBS->HOT_SHARD_THROTTLING_ENABLED && !pProxyCommitData->hotShards.empty()) {
+		if (self->checkHotShards()) {
+			TraceEvent(SevDebug, "ThrottledHotShard");
+			throw transaction_throttled_hot_shard();
```

Comment:
This is saying if there are any txns in the hot shard, we reject the whole batch. I think it should be only reject txns overlapping with the hot shard.

Reply:
To do that, I could mark the transaction aborted using `ConflictBatch::` , and would need to create a new enum: `ConflictBatch::TransactionThrottled`. It would be checked everywhere necessary, e.g. in the reply function, the way `TransactionTenantFailure`. Throttled transactions would not need to be checked for conflicts. 

That is complex. It may be simpler to move the hot shard check to post resolution. This way, a throttled transaction could be treated just like a conflict.

Reply:
What you propose can work. I haven't checked code (so unsure about complexity), but another idea is to "extract" these throttled txns out from the batch and reply errors early. Let the rest of the batch go through the pipeline.